### PR TITLE
ResultsCount support more result nodes

### PR DESCRIPTION
**Pull request by Akrion**
- Update ResultsCount to support the new node result type(s)

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>We were able to automatically fix some formatting issues in this PR for you!</td>
    </tr>
  </tbody>
</table>


### Some things that were possibly fixed:

- Code that could be fixed via the --fix flag
- Formatting that could be fixed by prettier

Take a look at this commit to see what happened in detail: ef1fa5ae865dd9c09fc64ee2c6b2c6caff91b9ac


And look at this wiki page to see the reasoning behind the ESLint rules: https://github.com/smartprocure/eslint-config-smartprocure/wiki/Rules-and-Why-We-Chose-Them
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 62b9ed77c35192811a48b353a58f65d4dfed2189
</p>

File: src/exampleTypes/ResultCount.js [hunk 1]
```diff
@@ -4,15 +4,32 @@ import { observer } from 'mobx-react'
...
+  noResults = 'No Results',
+}) => {
+  let count =
+    F.cascade(
+      [
+        'context.response.results.length',
+        'context.results.length',
+        'context.value',
+      ],
+      node
```
Comment (stellarhoof) [inline src/exampleTypes/ResultCount.js:19]: This sounds like a util to me

**Review by stellarhoof — APPROVED** [review]

File: src/exampleTypes/ResultCount.js [hunk 2]
```diff
@@ -4,15 +4,32 @@ import { observer } from 'mobx-react'
...
+    ) || _.isNumber(node.context)
```
Comment (daedalus28) [inline src/exampleTypes/ResultCount.js:20]: `node.context` should not be a scalar - what is this case for?

**Review by daedalus28 — CHANGES_REQUESTED** [review]

File: src/exampleTypes/ResultCount.js (hunk 2 above)
Comment (stellarhoof) [reply src/exampleTypes/ResultCount.js:20]: I made the new `count` node's context a scalar.

File: src/exampleTypes/ResultCount.js [hunk 3]
```diff
@@ -4,15 +4,32 @@ import { observer } from 'mobx-react'
...
   let totalRecords = count
-    ? F.cascade(['context.response.totalRecords', 'context.totalRecords'], node)
+    ? F.cascade(
+        [
+          'context.response.totalRecords',
+          'context.totalRecords',
+          'context.value',
+          'context',
```
[the remark this answers is not in the document]
Comment (Akrion) [reply src/exampleTypes/ResultCount.js:27]: Except the @stellarhoof `count` node added recently.

File: src/exampleTypes/ResultCount.js (hunk 2 above)
Comment (Akrion) [reply src/exampleTypes/ResultCount.js:20]: `count` is a scalar node. I asked about that before making this.

**Comment by Akrion** [comment]
> @Akrion @daedalus28 Since we're on this topic/component we should remove the inline loader. I've never seen a count with a loading style and if they have them that's just unnecessary spinners or whatever.

I actually had a real spinner when we are loading and it looked/felt better than having a stale number which changes at one point. Not sure but I think there are arguments for both cases.

File: src/exampleTypes/ResultCount.js (hunk 2 above)
Comment (daedalus28) [reply src/exampleTypes/ResultCount.js:20]: @stellarhoof we need to unwind that. The next major version will likely drop `context` as a wrapper entirely.

**Review by stellarhoof — APPROVED** [review]

File: src/exampleTypes/ResultCount.js (hunk 2 above)
Comment (stellarhoof) [reply src/exampleTypes/ResultCount.js]: Changed the returned value to `{ result: # }`

**Comment by Akrion** [comment]
Changed to the new `context.result`

**Review by ccalbaugh — APPROVED** [review]

**Pull request merged**